### DISCUSSION TITLE
f-droid package?

### DISCUSSION THREAD

**Discussion by IcyAlmond** [ppy/osu#13593] [Ideas] [answered]
Will this version of osu! be added onto f-droid?

**Comment by peppy** [comment, marked as answer]
Once we have things in a good state we will distribute on the google play store.

**Reply by orangixx** [reply]
Will a separate apk download still exist, and will that version be able to submit scores?

**Reply by peppy** [reply]
no idea

**Reply by chayleaf** [reply]
please do keep in mind not everyone uses Gapps, meaning an alternative is still necessary for some people

**Reply by dumbuz** [reply]
this is true, but people who don't use Gapps usually know they can either use different play store wrappers like Aurora Store or download from online mirrors (really not ideal though)

**Comment by ZeroAurora** [comment]
Seems not possible as F-Droid only accepts FOSS-only apks but osu! uses basslib

**Reply by orangixx** [reply]
https://github.com/ppy/osu-framework/issues/2784 *some day*

**Reply by bdach** [reply]
I wouldn't hold my breath if I were you. In fact, once the pending bassmix changes framework-side make it in, it'll be *less* likely that a switch to an alternative happens, not more.

**Reply by Hexality** [reply]
you can still make yourself a self-owned f-droid repo and add to the store, this way it will auto update if you allow it to.

**Reply by Diamochang** [reply]
Just because a library is not libre doesn't mean that the whole lazer can't be put on the shelf - just add a "software contains non-free parts" negative attribute to indicate that the `basslib` library is not libre. 

Always remember F-Droid isn't that _snarky_.

**Comment by chayleaf** [comment]
I created a repo, it's available at https://pavluk.org/fdroid/ and hosts official builds (I recommend installing the official build from Github first so I wouldn't be able to push a malicious apk with a different signature to you).

**Reply by bdach** [reply]
leaving aside the fact that you didn't even ask if it was fine to do this, i think it would be best if this "repository" was *clearly* marked as one that is not maintained by anyone here.

**Reply by chayleaf** [reply]
you're right, i'll make sure to build the apks myself then

**Reply by peppy** [reply]
please do NOT build them yourself. if you insist, make sure to remove all references to "ppy" or "osu!" from the project.

**Reply by peppy** [reply]
I'd highly suggest you first read all relevant discussions of other people doing this stuff because it's 99% being done wrong.

**Reply by chayleaf** [reply]
ok, i'll just mark it as unofficial in the package title

**Reply by chayleaf** [reply]
here's basically everything I did to set the repo up https://gist.github.com/pavlukivan/958d852ac4b84c532de671d355c2c7cf

from what I could tell the biggest issues were with people patching the binaries or building them from source, which I'm not doing in this case

**Reply by peppy** [reply]
I think that looks fine.

**Reply by Hexality** [reply]
It will be a while since I started to think about a F-Droid repo, but I was already about to lose my hopes.
I hope in the future you make a official F-Droid repository, cause as I said up there
> you can still make yourself a self-owned f-droid repo and add to the store, this way it will auto update if you allow it to.

Even if its not 100% FOSS, it's still a possible option since it's a self owned repo.
The good part about a F-Droid repo is that you can go and straight up add a button on a website/readme or a qr code that adds the repo to the f-droid store.

**Reply by ZeroAurora** [reply]
Providing an apk should be just enough - in my opinion. Setting up F-Droid repo is fine, but it might be unappealing to most players.

**Reply by Hexality** [reply]
The thing we should really have is a official Flathub package, since almost every single distro already have flatpak w/ Flathub installed by default.

**Reply by bdach** [reply]
@Hexality:

(a) what does flathub have to do with f-droid? one is linux, one is android
(b) https://github.com/flathub/flathub/pull/2619#issuecomment-973860402

**Reply by Hexality** [reply]
I know that, you just misunderstood what I meant.


Note that I said "every single >distro<"

**Reply by Hexality** [reply]
Btw I didn't saw that since I've been a bit out of the linux convo, but that was something that a bunch of people were wanting for a while so that's why I mentioned it.

**Comment by rollingmoai** [comment]
For the meanwhile, I am currently using [Obtainium](https://github.com/ImranR98/Obtainium) and it works great:
![image](https://user-images.githubusercontent.com/75471378/217862737-e287b70c-d91e-446a-8122-83ec4c26f9bd.png)

**Reply by Diamochang** [reply]
It's good, but Obtainium only lists the application and actually gets it installed by downloading it from GitHub, as described in their README:

>  Get Android app updates straight from the source.

Although this "app market" thinks it can replace F-Droid, I think jumping from the osu! official website to GitHub to download, and then using [App Manager](https://github.com/MuntashirAkon/AppManager) to display the app information can also achieve basically the same effect (the fundamental reason is that lazer now has a built-in updater).


If lazer is included in F-Droid, it can get the following benefits that Obtainium does **NOT** have:

1. Lazer can be listed alongside [opsu!](https://github.com/itdelatrisu/opsu), which has been listed before. This allows free software enthusiasts who only use F-Droid to know that osu! already has a liberalized **official** branch version, and can immediately choose whether to abandon the stable scoring method.

2. The installation package of lazer will be distributed intact and unmodified to F-Droid library mirrors around the world - this is very good news for players whose home network is poor or who are located in mainland China where GitHub is banned from time to time (because of programthink).

3. Lazer will have the opportunity to gain higher visibility in the Android free software community, as F-Droid is by far the leading application market for Android free software.

**Reply by tutacat** [reply]
Obtainium also has a search feature for searching github, gitlab, etc.
